### DISCUSSION TITLE
Different channels consume different queues serially.

### DISCUSSION THREAD

**Discussion by Yuan-zs** [rabbitmq/rabbitmq-java-client#1344] [Q&A] [answered]
hi all,
I found a strange phenomenon. For example, a connection creates 10 channels and then consumes 10 queues. according to the log printout, it turns out that these consumers are all using the same thread. I'd like to know what causes it. Thanks!

e.g
channel1 -> queue1
channel2 -> queue2
......
(to test, add `TimeUnit.SECONDS.sleep(10)` when consume)
the log like:
```
09:00:00 [pool-3-thread-3]  received 1 from queue1
09:00:10 [pool-3-thread-3]  received 1 from queue2
09:00:20 [pool-3-thread-3]  received 1 from queuex
...
```

It looks like the thread pool do not create new threads. The thread pool is generated by amqp-client when the connection is created, and there is nothing that can be changed to it.  I did not set custom thread pool when creating conn. I'm so confused. Whether there are other factors affecting the thread pool? I try to increasing cpu/memory, but doesn't work

version:
java 11
springboot 2.7.5
amqp-client 5.16.0

**Comment by lukebakken** [comment]
You should use multiple connections if you wish to use multiple threads (cc @acogoluegnes). At least, that's how the .NET client works.

**Reply by acogoluegnes** [reply, marked as answer]
The Java client uses a "work service" to make sure inbound messages on a given channel are processed in sequence (whatever the consumer on this channel). This work service uses a thread pool with `Runtime.getRuntime().availableProcessors()` threads by default, but this can be overridden by providing an `ExecutorService` to the `ConnectionFactory#newConnection()` method you are using.

The behavior you are experiencing should not happen if the thread pool is large enough and the consumers _are registered on different channels_. Please provide a standalone program or project to help us investigate if the problem persists.

**Reply by Yuan-zs** [reply]
Thanks for your reply, I got it. After investigation, that may be caused by the test env. And i closed this disscussion. thanks!
